core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.polling;
+
+import reactor.core.Disposable;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * This class provides implementation of long running operations. The poller starts polling <b>automatically</b> when Poller instance is created.
+ * It uses {@link Flux} from reactive programming model to achieve auto polling.
+ * It has function for usual operation of a poller. For example listen/observe poll responses, enable/disable auto polling,
+ * manual polling, wait for polling to complete and get status of current polling.
+ *
+ * <p><strong>Code Samples</strong></p>
+ *
+ * <p><strong>Instantiating and Subscribing to Poll Response</strong></p>
+ *
+ * <pre>
+ *     // Assumption : Our task is to create Certificate and the return CreateCertificateResponse in Poll Response. CreateCertificateResponse is user defined custom class.
+ *
+ * <p>     class CreateCertificateResponse {
+ *              String myresponse;
+ *              CreateCertificateResponse(String respone) {
+ *                  this.myresponse = respone;
+ *              }
+ *              public String toString() {
+ *                  return this.myresponse;
+ *              }
+ *          }
```

Comment:
Speak to Srikanta about externalising these code samples - might as well do that from the start.

Reply:
resolved